src/main/java/org/breedinginsight/services/ExperimentalCollaboratorService.java

```diff
@@ -31,4 +36,18 @@ public class ExperimentalCollaboratorService {
     public ExperimentalCollaboratorService(ExperimentalCollaboratorDAO experimentalCollaboratorDAO) {
         this.experimentalCollaboratorDAO = experimentalCollaboratorDAO;
     }
+
+    public ExperimentProgramUserRoleEntity createExperimentalCollaborator(UUID programUserRoleId, UUID experimentId, UUID userId) {
```

Comment:
Will there be a need to wrap `ExperimentProgramUserRoleEntity` into our own object for the purposes of returning via the api. I think sometimes we do that for serialization purposes, not sure if that will come up here.

Reply:
The need could arise, I don't think we would wrap ExperimentProgramUserRoleEntity, but rather extend/inherit it and swapping in the subclass for the parent class shouldn't be a major effort.

As an example, I'm looking at `ProgramUpload<T>`, but that seems like a special case, a lot of our DAOs use the jooq generated entity (which all implement `Serializable` FWIW) such as `BreedingMethodDAOImpl` methods returning the jooq generated `ProgramBreedingMethodEntity`.

Reply:
Ya sorry intended to mean extend, not compose. Agree we can just deal with it if the need arises.